slalombuild/secureli
# Feature/secureli 460 custom regex

secureli-460

Adds new `new-pattern` parameter to `secureli update` and updates `secureli scan` to check for custom regex


## Changes

*

## Testing

* run `secureli update --new-pattern <custom-regex-pattern>`
* verify that your custom regex pattern was added to `.secureli.yaml`
* modify any file (other than .secureli.yaml) to include text that matches your custom regex
* stage the changes
* run `secureli scan` and verify the scan fails
* revert the changes and run `secureli scan` and verify the scan passes

## Clean Code Checklist

- [x] Meets acceptance criteria for issue
- [x] New logic is covered with automated tests
- [ x Appropriate exception handling added
- [x] Thoughtful logging included
- [x] Follow-up work is documented in TODOs
- [x] TODOs have a ticket associated with them
- [x] No commented-out code included
